app/services/search_indexer.rb

```diff
@@ -53,15 +36,39 @@ def self.update_index(table: , id: , raw_data:)
 
     indexed_data = search_data.select { |d| d.length > 0 }.join(' ')
 
-    params = {
+    ranked_params = {
       a: search_data[0],
       b: search_data[1],
       c: search_data[2],
       d: search_data[3],
+    }
+
+    tsvector = DB.query_single("SELECT #{ranked_index}", ranked_params)[0]
+    additional_lexemes = []
+
+    tsvector.scan(/'(([a-zA-Z0-9]+\.)+[a-zA-Z0-9]+)'\:([\w+,]+)/).reduce(additional_lexemes) do |array, (lexeme, _, positions)|
+      count = 0
+
+      loop do
+        count += 1
+        break if count >= 10 # Safeguard here to prevent infinite loop when a term has many dots
```

Comment:
Realized we need a safe guard here to prevent extreme cases. 'some.super.long.file.name.that.will.never.end.some.super.long.file.name.that.will.never.end.some.super.long.file.name.that.will.never.end.some.super.long.file.name.that.will.never.end`